AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessor.swift

```diff
@@ -90,6 +90,8 @@ final class ItemListProcessor: StateProcessor<ItemListState, ItemListAction, Ite
                       let totpKey = TOTPKeyModel(authenticatorKey: key)
                 else { return }
                 await generateAndCopyTotpCode(totpKey: totpKey)
+            case .syncError:
+                break // no action for this type
```

Comment:
I had been looking at some of these and thinking we do the same thing over and over. So I pulled the bulk of the switch statements into a computed property on ItemListItem (with tests) so that most places we can just do one `if let` and get the TOTP code. So that reduced the overall surface area of us having to switch. I've also tried adding some more tests to as many cases as I could. So I think we're in a much better spot.